EchoPro/survey.py

```diff
@@ -36,741 +25,457 @@ class Survey:
-
-        if not params:
-            raise ValueError("You must provide parameters for the semi-variogram!")
-
-        # make sure all parameters are included
-        if set(vario_type_dict.keys()).difference(set(params.keys())):
-            raise ValueError("Some required parameters where not provided!")
-
-        # check that all types are correct for params
-        for key, val in params.items():
-            expected_type = vario_type_dict.get(key)
-            if not isinstance(val, expected_type):
-                raise TypeError(f"{key} is not of type {expected_type}")
-
-        # provide a warning if the transect_results_gdf being used was
-        # created from a subset of the full data
-        if (len(self.bio_calc.transect_results_gdf) != len(self.nasc_df)) and warning:
-            warn(
-                "The biomass data being used is a subset of the full dataset. "
-                "It is recommended that you use the biomass data created from the full dataset. "
-                "To silence this warning set the warning argument to False."
-            )
-
-        if (not isinstance(self.bio_calc.transect_results_gdf, gpd.GeoDataFrame)) and (
-            "biomass_density_adult" not in self.bio_calc.transect_results_gdf
-        ):
-            raise ValueError(
-                "The areal biomass density must be calculated before running this routine!"
+        # Error evaluation, if applicable
+        if param_intersect:
+            raise RuntimeError(
+                f"The initialization and survey year configuration files comprise the following intersecting variables: {param_intersect}"
             )
 
-        semi_vario = SemiVariogram(
-            krig_mesh.transformed_transect_df.x_transect.values,
-            krig_mesh.transformed_transect_df.y_transect.values,
-            self.bio_calc.transect_results_gdf[
-                "biomass_density_adult"
-            ].values.flatten(),
-            params["lag_res"],
-            params["nlag"],
-        )
+        ### Format dictionary that will parameterize the `config` class attribute
+        # Join the initialization and survey year parameters into a single dictionary
+        full_params = {**init_config_params, **survey_year_config_params}
 
-        return semi_vario
-
-    def get_kriging(self, params: krig_param_type) -> Kriging:
+        # ??? TODO: Since this is a static method, this carriage return may not be the most appropriate way to pass 
+        # 'full_params' to the class instance
+        return full_params
+    
+    def search_nested_dict( self ,
+                            config: dict ,
+                            filename: str ,
+                            current_path: list = [] ):
         """
-        Initializes a ``Kriging`` object using the
-        provided parameters
+        Recursive utility function that searches the configuration attribute
+        (a nested dictionary) for a specific 
```

Comment:
Since the config dictionaries are clearly defined and under our control, the use of recursive function seems a bit overkill -- I think writing functions to search for specific files would suffice, especially since this function is used specifically for finding where a path is located, which is very specific already.

Reply:
See my other more overarching comment https://github.com/uw-echospace/EchoPro/pull/161#discussion_r1443614042